### DISCUSSION TITLE
Create Mock of a Riverpod-generated AsyncNotifierProvider with freezed Model

### DISCUSSION THREAD

**Discussion by Takrem1** [rrousselGit/riverpod#2711] [Q&A] [answered]
I have a Riverpod generated AsyncNotifierProvider class that holds an ObjectModel created by Freezed. I now need to create a mock of the Riverpod class for my unit tests. I am currently doing this:
```dart
class MockMyFreezedModelProvider extends AsyncNotifierProviderElement<MyFreezedModelProvider, MyFreezedModel>
    with Mock
    implements MyFreezedModelProvider {}
```

I added the `extends AsyncNotifierProviderElement<MyFreezedModelProvider, MyFreezedModel>` because otherwise I get the error message `Class 'MockMyFreezedModelProvider' has no instance method '_setElement'.get`

However, now I get the following error message:

```
Superinterfaces don't have a valid override for 'update': ProviderElementBase.update (void
Function(ProviderBase<AsyncValue<MyFreezedModel>>)), AsyncNotifierBase.update (Future<MyFreezedModel>
Function(FutureOr<MyFreezedModel> Function(MyFreezedModel), {FutureOr<MyFreezedModel> Function(Object, StackTrace)?
onError})). (Documentation)
```

Any idea how to solve this?

My freezed Model:
```dart
@freezed
class MyFreezedModel with _$MyFreezedModel {
  const factory MyFreezedModel({
    required bool test, // simplified
  }) = _MyFreezedModel;

  factory MyFreezedModel.fromJson(Map<String, dynamic> json) => _$MyFreezedModelFromJson(json);
}
```

**Comment by rrousselGit** [comment]
You should do `extends _$MyNotifer with Mock implements MyNotifier`

**Reply by Takrem1** [reply]
With this:

```dart
class MockMyFreezedModelProvider extends _$MyFreezedModelProvider with Mock implements MyFreezedModelProvider {}
```
I get the following error:
`Mixin can only be applied to class.`

**Reply by Takrem1** [reply, marked as answer]
I found a solution. I had to use `extends AsyncNotifier<MyFreezedModel>` because _$MyFreezedModelProvider was only a typedef `typedef _$MyFreezedModelProvider = AsyncNotifier<MyFreezedModel>;`

**Reply by rrousselGit** [reply]
That shouldn't be a problem. Do you maybe have a full example I could copy-paste to reproduce this error?
